File: backend/src/main/java/com/trade/service/AuthService.java
package com.trade.service;

import com.trade.dto.auth.AuthResponse;
import com.trade.dto.auth.LoginRequest;
import com.trade.dto.auth.ProfileUpdateRequest;
import com.trade.dto.auth.RegisterRequest;
import com.trade.dto.auth.UserProfileResponse;

/**
 * Contract for authentication and user profile operations.
 */
public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    UserProfileResponse getProfile(String email);

    UserProfileResponse updateProfile(ProfileUpdateRequest request, String email);

    boolean isEmailAvailable(String email);

    boolean isPhoneAvailable(String phone);
}

File: frontend/src/services/authService.js
/**
 * authService.js — Authentication API endpoints.
 *
 * Backend: AuthController @ /api/auth
 * Endpoints: register, login, profile, check-email, check-phone
 */

import { get, post, put } from './httpClient';

export const authApi = {
  /**
   * POST /api/auth/register
   * @param {object} data - { firstName, lastName, email, phone, password, role, company, country, ... }
   */
  register(data) {
    return post('/auth/register', data);
  },

  /**
   * POST /api/auth/login
   * @param {{email: string, password: string}} credentials
   */
  login(credentials) {
    return post('/auth/login', credentials);
  },

  /**
   * GET /api/auth/profile — current user profile
   */
  getProfile() {
    return get('/auth/profile');
  },

  /**
   * PUT /api/auth/profile — update current user profile
   * @param {object} data
   */
  updateProfile(data) {
    return put('/auth/profile', data);
  },

  /**
   * GET /api/auth/check-email?email=...
   * Real-time duplicate detection (debounced on frontend).
   */
  checkEmail(email) {
    return get(`/auth/check-email?email=${encodeURIComponent(email)}`);
  },

  /**
   * GET /api/auth/check-phone?phone=...
   * Real-time duplicate detection (debounced on frontend).
   */
  checkPhone(phone) {
    return get(`/auth/check-phone?phone=${encodeURIComponent(phone)}`);
  },
};
